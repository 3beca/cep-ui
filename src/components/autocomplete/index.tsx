import * as React from 'react';
import MUIAutocomplete, {createFilterOptions} from '@material-ui/lab/Autocomplete';
import TextField from '@material-ui/core/TextField';
import CircularProgress from '@material-ui/core/CircularProgress';
import { EntityWithName } from '../../services/api';
import { FilterOptionsState } from '@material-ui/lab/useAutocomplete';
import {useDebounce} from '../../services/use-debounce';

export const createCustomFilterOptions = createFilterOptions;

export type AutocompleteProps<T> = {
    disabled?: boolean;
    options: T[];
    selected: T|null;
    setSelected(element: T|null): void;
    isLoading: boolean;
    emptyElement: T;
    changeFilter(filter: string): void;
    defaultFilter?: (options: T[], state: FilterOptionsState<T>) => T[];
    ariaLabel?: string;
    elementId?: string;
    placeholderText?: string;
    loadingText?: string;
    className?: string;
};
export function Autocomplete<T extends EntityWithName>(
    {
        disabled = false,
        options,
        selected,
        setSelected,
        isLoading,
        emptyElement,
        changeFilter,
        defaultFilter = createFilterOptions<T>(),
        ariaLabel = 'element name',
        elementId = 'element-autocomplete',
        placeholderText = 'Search a element',
        loadingText = 'Loading elements',
        className
    }: AutocompleteProps<T>) {
    const cameFromLoading = React.useRef(false);
    const handleChange = React.useCallback(
        (ev: React.ChangeEvent<unknown>, element: T|null) => {
            setSelected(element);
        },[setSelected]
    );
    const handleGetOptions = React.useCallback(
        (option: T) => {
            return option.name;
        }, []
    );
    const handleRenderOptions = React.useCallback(
        (option: T) => {
            return (option.id !== emptyElement.id ? '' : 'Create ') + option.name;
        }, [emptyElement]
    );
    const handleFilterOptions = React.useCallback(
        (options: T[], params: FilterOptionsState<T>) => {
            const filtered = defaultFilter(options, params);
            // If there are not matchs...
            if (cameFromLoading.current && !isLoading && options.length === 0 && filtered.length === 0 && params.inputValue !== '') {
                const newElement = {...emptyElement, name: params.inputValue};
                filtered.push(newElement);
            }
            return filtered;
        }, [isLoading, emptyElement, defaultFilter]
    );
     // Debounce filter
     const [filter, setFilter] = useDebounce({callback: changeFilter, initialValue: ''});
     const handleInputChange = React.useCallback(
        (ev: React.ChangeEvent<unknown>, name: string) => {
            setFilter(name);
        },[setFilter]
    );
    React.useEffect(() => {
        cameFromLoading.current = isLoading;
    });

    return (
        <MUIAutocomplete
            disabled={disabled}
            autoComplete={true}
            fullWidth={true}
            id={elementId}
            aria-label={ariaLabel}
            multiple={false}
            value={selected}
            className={className}
            options={options}
            filterOptions={handleFilterOptions}
            getOptionLabel={handleGetOptions}
            renderOption={handleRenderOptions}
            onChange={handleChange}
            loading={isLoading}
            onInputChange={handleInputChange}
            inputValue={filter}
            renderInput={
                (params) => {
                    return (
                        <TextField
                            {...params}
                            label={isLoading ? loadingText : placeholderText}
                            InputProps={
                                {
                                ...params.InputProps,
                                endAdornment: (
                                    <React.Fragment>
                                        {isLoading ? <CircularProgress color="primary" size={20} /> : null}
                                        {params.InputProps.endAdornment}
                                    </React.Fragment>
                                )
                                }
                            }
                        />
                    )
                }
            }
        />
    );
};

export default Autocomplete;
