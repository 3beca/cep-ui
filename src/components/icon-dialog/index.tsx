import * as React from 'react';
import IconButton from '@material-ui/core/IconButton';
import {NOOP} from '../../utils';
import Dialog from '@material-ui/core/Dialog';

type IconDialogContext = ()=>void;
const IconDialogContext = React.createContext<IconDialogContext>(NOOP);
export const useIconDialog = () => {
    return React.useContext<IconDialogContext>(IconDialogContext);
};

export type IconDialogProps = {show: boolean; icon: React.ReactElement; onOpen?():void, onClose?():void};
export const IconDialog: React.FC<IconDialogProps> = ({show, icon, onOpen = NOOP, onClose = NOOP, children, ...props}) => {
    const [open, setOpen] = React.useState(false);
    const closeDialog = React.useCallback(() => {
        setOpen(false);
        onClose();
    }, [onClose]);
    const openDialog = React.useCallback(() => {
        setOpen(true);
        onOpen();
    }, [onOpen]);
    if(!open && !show) return null;
    if(open) {
        return (
            <Dialog
                open={open}
                onClose={closeDialog}
                id='icon-dialog'
                scroll='paper'
                aria-labelledby='icon-dialog-title'
                aria-describedby='icon-dialog-content'>
                <IconDialogContext.Provider value={closeDialog} children={children}/>
            </Dialog>
        );
    }
    return (
        <IconButton aria-label='open dialog' onClick={openDialog} {...props}>{icon}</IconButton>
    );
};

export default IconDialog;

