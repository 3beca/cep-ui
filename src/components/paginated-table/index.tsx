import React from 'react';
import TableContainer from '@material-ui/core/TableContainer';
import TablePagination from '@material-ui/core/TablePagination';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import {NOOP} from '../../utils';

import { useStyles } from './styles';

const TableLoadingView: React.FC<{show: boolean;}> = ({show}) => {
    const styles = useStyles();
    if (!show) return null;
    return (
        <div
            data-testid='loading-view-row' className={styles.loadingView}>
            <div>
                <CircularProgress className={styles.loadingSpinner} />
            </div>
        </div>
    );
};

const TableEmptyView: React.FC<{show: boolean;}> = ({show}) => {
    const styles = useStyles();
    if (!show) return null;
    return (
        <div
            data-testid='empty-view-row' >
            <div>
                <Typography align='center' className={styles.errorView}>No Elements created yet!</Typography>
            </div>
        </div>
    );
};
export type RowsSizes = 5|10|20;
export type PaginatedTableProps = {
    page?: number;
    size?: RowsSizes;
    isLoading?: boolean;
    isEmpty?: boolean;
    hasNextPage?: boolean;
    hasPrevPage?: boolean;
    onChangePage?(page: number):void;
    onChangePageSize?(page: RowsSizes):void;
};
export const PaginatedTable: React.FC<PaginatedTableProps> = React.memo(
    ({
        page = 1,
        size = 10,
        isLoading = true,
        isEmpty = true,
        hasNextPage = false,
        hasPrevPage = false,
        onChangePage=NOOP,
        onChangePageSize=NOOP,
        children
    }) => {
        return (
            <>
                <TableContainer>
                   {children}
                </TableContainer>
                <TableEmptyView show={!isLoading && isEmpty}/>
                <TableLoadingView show={true}/>
                <TablePagination
                    role='paginator'
                    component='div'
                    rowsPerPageOptions={[5, 10, 20]}
                    count={-1}
                    rowsPerPage={size}
                    page={page - 1}
                    nextIconButtonProps={{disabled: !hasNextPage}}
                    backIconButtonProps={{disabled: !hasPrevPage}}
                    onChangePage={(ev, page) => onChangePage(page + 1)}
                    onChangeRowsPerPage={(ev) => onChangePageSize(Number(ev.target.value) as RowsSizes)}
                    SelectProps={{'aria-label': 'pageSelector'}}
                />
            </>
        );
    }
);

export default PaginatedTable;
