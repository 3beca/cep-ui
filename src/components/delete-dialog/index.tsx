import * as React from 'react';
import Typography from '@material-ui/core/Typography';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Button from '@material-ui/core/Button';
import LinearProgress from '@material-ui/core/LinearProgress';

import {useIconDialog} from '../../components/icon-dialog';
import { EventType, ServiceError, Target, ServiceDeleted } from '../../services/api';
import { useDelete, ENTITY } from '../../services/api/use-api';
import {useStyles} from './styles';

const EmptyMessage: React.FC<{show: boolean;}> = ({show}) => {
    if(!show) return null;
    return (
        <div aria-label='empty message'>
            <Typography variant='h6'>There are NOT Elements selected to be deleted!</Typography>
        </div>
    );
};

const ErrorMessage: React.FC<{error: ServiceError|undefined;}> = ({error}) => {
    if(!error) return null;
    return (
        <div aria-label='error message'>
            <Typography variant='h3'>{error.message}</Typography>
        </div>
    );
};

const Actions: React.FC<{close():void; disableClose: boolean; disableDelete: boolean, deleteElements():void}> = ({close, disableClose, disableDelete, deleteElements}) => {
    const styles = useStyles();
    return (
        <DialogActions aria-label='actions'>
            <Button onClick={close} disabled={disableClose}>Close</Button>
            <Button onClick={deleteElements} disabled={disableDelete} className={styles.deleteButton}>Delete</Button>
        </DialogActions>
    );
};

const LoadingView: React.FC<{show: boolean}> = ({show}) => {
    if (!show) return (
        <div aria-label='element loader'>
            <LinearProgress variant='determinate' value={100} color='primary'/>
        </div>
    );
    return (
        <div aria-label='deleting element'>
            <LinearProgress variant='indeterminate' color='secondary'/>
        </div>
    );
};

export type ELEMENTS = (EventType|Target)[];
const DeleteList: React.FC<{show: boolean; elements?: ELEMENTS; isLoading: boolean}> = ({show, elements, isLoading}) => {
    const styles = useStyles();
    if (!show) return null;
    isLoading = true;
    return (
        <div aria-label='elements to delete' className={styles.elementList}>
        {
            elements!.map(
                (e) => (
                    <div key={e.id} aria-label='element to delete' className={styles.elementItem}>
                        <div><Typography variant='h5'>{e.name}</Typography></div>
                        <div><Typography variant='caption' className={styles.captionText}>{e.id}</Typography></div>
                    </div>
                )
            )
        }
        </div>
    );
};

const DeletedList: React.FC<{responses: ServiceDeleted[]|undefined; elements: ELEMENTS|undefined;}> = ({responses, elements}) => {
    const styles = useStyles();
    if (!Array.isArray(elements) || elements.length === 0) return null;
    if (!Array.isArray(responses) || responses.length === 0) return null;
    return (
        <div aria-label='elements deleted' className={styles.elementList}>
        {
            responses.map(
                (event, idx) => {
                    //if(event.state === 'DELETED') return null;
                    return (
                        <div key={event.id} aria-label='element to delete' className={styles.elementItem}>
                            <div aria-label='deleted element'><Typography variant='h5'>{elements[idx].name}</Typography></div>
                            {
                                event.state === 'DELETED' ?
                                (<div aria-label='success message'><Typography variant='caption' className={styles.successText}>This entity has been deleted successfuly</Typography></div>) :
                                (<div aria-label='error message'><Typography variant='caption' className={styles.errorText}>Error: {event.error ? event.error.message : 'This element cannot be deleted'}</Typography></div>)
                            }
                        </div>
                    );
                }
            )
        }
        </div>
    );
};

export type DeleteDialogProps = {title: string; entity: ENTITY; elementsSelecteds?: ELEMENTS; onDeleted?():void;};
export const DeleteDialog: React.FC<DeleteDialogProps> = React.memo(({title, entity, elementsSelecteds, onDeleted}) => {
    const styles = useStyles();
    const [elements] = React.useState(elementsSelecteds);
    const closeDialog = useIconDialog();
    const events = React.useMemo(() => elements ? elements?.map(e => e.id) : [], [elements])
    const {isLoading, response, error, request} = useDelete(entity, events);
    const hasElements = Array.isArray(elements) && elements.length > 0;
    const hasResponse = !(!Array.isArray(response?.data) || response?.data?.length === 0);
    React.useEffect(
        () => {
            // TODO: Si hay alguna respuesta con delete, recargar el listado de elements
            hasResponse && (typeof onDeleted === 'function') && onDeleted();
        }, [hasResponse, onDeleted]
    );
    return (
        <div className={styles.container}>
            <DialogTitle aria-label='title' id='icon-dialog-title'>
                {title}
            </DialogTitle>
            <DialogContent dividers={true} className={styles.dialogContent} id='icon-dialog-content'>
                <EmptyMessage show={!hasElements && !hasResponse && !error}/>
                <DeleteList show={hasElements && !hasResponse && !error} elements={elements} isLoading={isLoading}/>
                <DeletedList elements={elements} responses={response?.data}/>
                <ErrorMessage error={error && error.error}/>
            </DialogContent>
            <LoadingView show={isLoading}/>
            <Actions close={closeDialog} disableClose={isLoading} disableDelete={!hasElements || hasResponse || !!error} deleteElements={request}/>
        </div>
    );
}, () => true);

export default DeleteDialog;

