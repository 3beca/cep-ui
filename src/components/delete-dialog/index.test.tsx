import * as React from 'react';
import {
    renderWithAPI as render,
    screen
} from '../../test-utils';
import userEvent from '@testing-library/user-event';
import {generateEventTypeListWith, setupNock, serverDeleteEventType} from '../../test-utils';
import {useIconDialog} from '../../components/icon-dialog';
import {BASE_URL} from '../../services/config';
import DeleteDialog from './index';
import { EventType } from '../../services/api';
import { ENTITY } from '../../services/api-provider/use-api';

jest.mock('../../components/icon-dialog', () => {
    const mockClose = jest.fn();
    return {
        useIconDialog: () => mockClose,
        mockClose
    };
});
const mockUseIconDialog = useIconDialog as unknown as () => jest.Mock;
afterEach(() => mockUseIconDialog().mockClear());

test('Should render a dialog with empty message and snapshot', async () => {
    const closeCallback = mockUseIconDialog();
    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES}/>);
    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/empty message/i);
    expect(screen.queryByLabelText(/elements to delete/i)).not.toBeInTheDocument();
    expect(screen.queryByLabelText(/^delete button$/i)).toBeDisabled();
    const closeButton = await screen.findByLabelText(/close button/i);

    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});

test('Should render a dialog with a list of EventTypes to delete and snapshot', async () => {
    const eventTypeList = generateEventTypeListWith(3, false, false);
    const closeCallback = mockUseIconDialog();
    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES} elementsSelecteds={eventTypeList.results}/>);

    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/elements to delete/i);
    expect(screen.queryByLabelText(/empty message/i)).not.toBeInTheDocument();
    const closeButton = await screen.findByLabelText(/close button/i);

    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});

test('Should render a dialog with a list of EventTypes to delete and delete all items and snapshot', async () => {
    const eventTypeList = generateEventTypeListWith(3, false, false);
    const closeCallback = mockUseIconDialog();
    const onDelete = jest.fn();

    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES} elementsSelecteds={eventTypeList.results}  onDeleted={onDelete}/>);

    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/elements to delete/i);
    expect(screen.queryByLabelText(/empty message/i)).not.toBeInTheDocument();
    const closeButton = await screen.findByLabelText(/close button/i);
    const deleteButton = await screen.findByLabelText(/^delete button$/i);

    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[0].id, 200);
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[1].id, 200);
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[2].id, 200);
    userEvent.click(deleteButton);

    await screen.findByLabelText(/deleting element/);

    expect(await screen.findAllByLabelText(/deleted element/)).toHaveLength(3);
    expect(await screen.findAllByLabelText(/success message/i)).toHaveLength(3);
    expect(deleteButton).toBeDisabled();
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onDelete).toHaveBeenNthCalledWith(1, [eventTypeList.results[0].id, eventTypeList.results[1].id, eventTypeList.results[2].id]);

    // Close dialog
    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});

test('Should render a dialog with a list of 4 EventTypes to delete and fail to delete 2 and snapshot', async () => {
    const eventTypeList = generateEventTypeListWith(4, false, false);
    const closeCallback = mockUseIconDialog();
    const onDelete = jest.fn();

    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES} elementsSelecteds={eventTypeList.results} onDeleted={onDelete}/>);

    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/elements to delete/i);
    expect(screen.queryByLabelText(/empty message/i)).not.toBeInTheDocument();
    const closeButton = await screen.findByLabelText(/close button/i);
    const deleteButton = await screen.findByLabelText(/^delete button$/i);

    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[0].id, 200);
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[1].id, 500, {error: 'invalid id', message: 'cannot delete eventtype', statusCode: 400});
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[2].id, 400);
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[3].id, 200);
    userEvent.click(deleteButton);

    await screen.findByLabelText(/deleting element/);

    expect(await screen.findAllByLabelText(/deleted element/)).toHaveLength(4);
    expect(await screen.findAllByLabelText(/error message/i)).toHaveLength(2);
    expect(await screen.findAllByLabelText(/success message/i)).toHaveLength(2);
    expect(deleteButton).toBeDisabled();
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onDelete).toHaveBeenNthCalledWith(1, [eventTypeList.results[0].id, eventTypeList.results[3].id]);
    // Close dialog
    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});

test('Should render a dialog with a list of 2 EventTypes to delete and fail to delete all', async () => {
    const eventTypeList = generateEventTypeListWith(2, false, false);
    const closeCallback = mockUseIconDialog();
    const onDelete = jest.fn();

    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES} elementsSelecteds={eventTypeList.results} onDeleted={onDelete}/>);

    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/elements to delete/i);
    expect(screen.queryByLabelText(/empty message/i)).not.toBeInTheDocument();
    const closeButton = await screen.findByLabelText(/close button/i);
    const deleteButton = await screen.findByLabelText(/^delete button$/i);

    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[0].id, 500, {error: 'invalid id', message: 'cannot delete eventtype', statusCode: 400});
    serverDeleteEventType(setupNock(BASE_URL), eventTypeList.results[1].id, 500, {error: 'invalid id', message: 'cannot delete eventtype', statusCode: 400});
    userEvent.click(deleteButton);

    await screen.findByLabelText(/deleting element/);

    expect(await screen.findAllByLabelText(/deleted element/)).toHaveLength(2);
    expect(await screen.findAllByLabelText(/error message/i)).toHaveLength(2);
    expect(screen.queryAllByLabelText(/success message/i)).toHaveLength(0);
    expect(deleteButton).toBeDisabled();
    expect(onDelete).toHaveBeenCalledTimes(0);
    // Close dialog
    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});

test('Should render a dialog with error when evettypes is invalid and snapshot', async () => {
    const closeCallback = mockUseIconDialog();
    const onDelete = jest.fn();
    const invalidEventType = {id: null} as unknown as EventType;
    render(<DeleteDialog title='Dialog title' entity={ENTITY.EVENT_TYPES} elementsSelecteds={[invalidEventType]} onDeleted={onDelete}/>);

    await screen.findByLabelText(/title/);
    await screen.findByLabelText(/actions/);
    await screen.findByLabelText(/elements to delete/i);
    expect(screen.queryByLabelText(/empty message/i)).not.toBeInTheDocument();
    const closeButton = await screen.findByLabelText(/close button/i);
    const deleteButton = await screen.findByLabelText(/^delete button$/i);

    userEvent.click(deleteButton);
    await screen.findByLabelText(/error message/i);
    expect(onDelete).toHaveBeenCalledTimes(0);

    userEvent.click(closeButton);
    expect(closeCallback).toHaveBeenCalledTimes(1);
});