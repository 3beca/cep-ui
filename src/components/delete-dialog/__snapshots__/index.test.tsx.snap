// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a dialog with a list of 4 EventTypes to delete and fail to delete 2 and snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-322"
  >
    <div
      aria-label="title"
      class="MuiDialogTitle-root"
      id="icon-dialog-title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Dialog title
      </h2>
    </div>
    <div
      class="MuiDialogContent-root makeStyles-dialogContent-324 MuiDialogContent-dividers"
      id="icon-dialog-content"
    >
      <div
        aria-label="elements deleted"
        class="makeStyles-elementList-325"
      >
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-327"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType0
            </h5>
          </div>
          <div
            aria-label="success message"
          >
            <span
              class="MuiTypography-root makeStyles-successText-330 MuiTypography-caption"
            >
              This entity has been deleted successfuly
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-327"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType1
            </h5>
          </div>
          <div
            aria-label="error message"
          >
            <span
              class="MuiTypography-root makeStyles-errorText-329 MuiTypography-caption"
            >
              Error: 
              cannot delete eventtype
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-327"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType2
            </h5>
          </div>
          <div
            aria-label="error message"
          >
            <span
              class="MuiTypography-root makeStyles-errorText-329 MuiTypography-caption"
            >
              Error: 
              This element cannot be deleted
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-327"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType3
            </h5>
          </div>
          <div
            aria-label="success message"
          >
            <span
              class="MuiTypography-root makeStyles-successText-330 MuiTypography-caption"
            >
              This entity has been deleted successfuly
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-label="element loader"
    >
      <div
        aria-valuenow="100"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(0%);"
        />
      </div>
    </div>
    <div
      aria-label="actions"
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Close
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-deleteButton-323 Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render a dialog with a list of EventTypes to delete and delete all items and snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-215"
  >
    <div
      aria-label="title"
      class="MuiDialogTitle-root"
      id="icon-dialog-title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Dialog title
      </h2>
    </div>
    <div
      class="MuiDialogContent-root makeStyles-dialogContent-217 MuiDialogContent-dividers"
      id="icon-dialog-content"
    >
      <div
        aria-label="elements deleted"
        class="makeStyles-elementList-218"
      >
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-220"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType0
            </h5>
          </div>
          <div
            aria-label="success message"
          >
            <span
              class="MuiTypography-root makeStyles-successText-223 MuiTypography-caption"
            >
              This entity has been deleted successfuly
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-220"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType1
            </h5>
          </div>
          <div
            aria-label="success message"
          >
            <span
              class="MuiTypography-root makeStyles-successText-223 MuiTypography-caption"
            >
              This entity has been deleted successfuly
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-220"
        >
          <div
            aria-label="deleted element"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType2
            </h5>
          </div>
          <div
            aria-label="success message"
          >
            <span
              class="MuiTypography-root makeStyles-successText-223 MuiTypography-caption"
            >
              This entity has been deleted successfuly
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-label="element loader"
    >
      <div
        aria-valuenow="100"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(0%);"
        />
      </div>
    </div>
    <div
      aria-label="actions"
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Close
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-deleteButton-216 Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render a dialog with a list of EventTypes to delete and snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-108"
  >
    <div
      aria-label="title"
      class="MuiDialogTitle-root"
      id="icon-dialog-title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Dialog title
      </h2>
    </div>
    <div
      class="MuiDialogContent-root makeStyles-dialogContent-110 MuiDialogContent-dividers"
      id="icon-dialog-content"
    >
      <div
        aria-label="elements to delete"
        class="makeStyles-elementList-111"
      >
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-113"
        >
          <div>
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType0
            </h5>
          </div>
          <div>
            <span
              class="MuiTypography-root makeStyles-captionText-114 MuiTypography-caption"
            >
              0__3
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-113"
        >
          <div>
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType1
            </h5>
          </div>
          <div>
            <span
              class="MuiTypography-root makeStyles-captionText-114 MuiTypography-caption"
            >
              1__3
            </span>
          </div>
        </div>
        <div
          aria-label="element to delete"
          class="makeStyles-elementItem-113"
        >
          <div>
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Elemento EventType2
            </h5>
          </div>
          <div>
            <span
              class="MuiTypography-root makeStyles-captionText-114 MuiTypography-caption"
            >
              2__3
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-label="element loader"
    >
      <div
        aria-valuenow="100"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(0%);"
        />
      </div>
    </div>
    <div
      aria-label="actions"
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Close
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-deleteButton-109"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Delete
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render a dialog with empty message and snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <div
      aria-label="title"
      class="MuiDialogTitle-root"
      id="icon-dialog-title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Dialog title
      </h2>
    </div>
    <div
      class="MuiDialogContent-root makeStyles-dialogContent-3 MuiDialogContent-dividers"
      id="icon-dialog-content"
    >
      <div
        aria-label="empty message"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          There are NOT Elements selected to be deleted!
        </h6>
      </div>
    </div>
    <div
      aria-label="element loader"
    >
      <div
        aria-valuenow="100"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(0%);"
        />
      </div>
    </div>
    <div
      aria-label="actions"
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Close
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-deleteButton-2 Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render a dialog with error when evettypes is invalid and snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-429"
  >
    <div
      aria-label="title"
      class="MuiDialogTitle-root"
      id="icon-dialog-title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Dialog title
      </h2>
    </div>
    <div
      class="MuiDialogContent-root makeStyles-dialogContent-431 MuiDialogContent-dividers"
      id="icon-dialog-content"
    >
      <div
        aria-label="error message"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          id is an invalid id value or array
        </h3>
      </div>
    </div>
    <div
      aria-label="element loader"
    >
      <div
        aria-valuenow="100"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(0%);"
        />
      </div>
    </div>
    <div
      aria-label="actions"
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Close
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-deleteButton-430 Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;
